### DISCUSSION TITLE
서버 수 관련

### DISCUSSION THREAD

**Discussion by JellyBrick** [koreanbots/koreanbots#431] [제안] [answered]
# 문제

## 들어가며

현재 `한국 디스코드 봇 리스트` 웹사이트는, 봇의 이용 서버 수가 많아도 아무런 혜택이 주어지지 않습니다. (추천 수가 같을 경우, 우선 순위가 달라지는 정도에서 그친다고 알고 있습니다)

## 자료 1

![자료 1](https://user-images.githubusercontent.com/16558115/122676943-1bf62c00-d21b-11eb-99ff-ee5f484e8228.png)
- 자료 1

`[자료 1]`/`KoreanBots` 사이트는 한국 디스코드 봇 사이트들 중 상당한 영향력을 가지고 있고,
추천 순위로 인해 상위에 노출되고 있는 봇은 이용자 수가 빠르게 증가하는 것을 볼 수 있습니다.

## 자료 2

![자료 2](https://user-images.githubusercontent.com/16558115/122677296-b9059480-d21c-11eb-92e9-0248087155b7.png)
- 자료 2

`[자료 2]`를 통해, 추천 수 기반 리스트는 봇의 이용 서버 수가 매우 많이 차이나도
추천 수가 1개라도 더 많다면, 순위가 올라간다는 것을 알 수 있습니다.

이로 인해서 형평성의 문제가 발생할 수 있습니다.

## 자료 3

![자료 3 / 출처: 전적이 봇 설명](https://user-images.githubusercontent.com/16558115/122677639-1e0dba00-d21e-11eb-8133-6baad62de687.png)

또한, `[자료 3]`에서 볼 수 있듯 몇몇 봇들은 추천을 하면 큰 혜택을 주게되는데,
이는 봇의 특성(`게임`/`노래`/`투표` 등등)에 따라, 비슷한 기능을 구현하더라도 받을 수 있는 추천 수가 크게 달라지기 때문에,
공정성을 해칠 수 있는 문제라고 생각됩니다.
(`자료 3 예` 참고)
(물론 해당 기능 자체는 해외의 몇몇 `디스코드 봇 랭킹 사이트`에서 허용하고 있는 바, 크게 잘못된 것은 아니라고 생각합니다)

### 자료 3 예

- `투표 봇 A`는 투표를 생성하는 것 외 별다른 기능이 없음. `랭킹 사이트`에서 추천을 하면 생성할 수 있는 투표 개수가 더 늘어나게 됨
- `게임 봇 B`는 `랭킹 사이트`에서 추천을 받으면, 게임 순위를 가를 수 있는 게임 머니를 주게 됨. (타 이용자와의 경쟁을 유발)

이는 "`디스코드 봇 리스트` 및 `디스코드` 이용자의 연령대" / "게임이라는 특성 상 경쟁성이 존재하는 점"을 고려,
(위에 기술했듯) 공정성을 해칠 수 있는 문제라고 생각됩니다.

# 대안

따라서 이에 대한 대책이 필요하다고 생각되는 바, 대안을 건의드립니다.

- 대안 1. 서버 수 기반 리스트 추가
 예전에도 비슷한 건의가 있었던 것으로 알고 있습니다.
하지만, 당시에 `서버 수 기반 리스트는 조작의 가능성이 있어서 제작을 하지 않을 것이다`라고 말하신 것으로 알고 있습니다.
디스코드 봇 초대 페이지에서 서버 수를 확인할 수 있는 점을 고려한다면, 이는 조작을 할 수 없는 구조라고 생각됩니다.

- 대안 2. 서버 수 + 추천 수 기반 리스트로 변경
 대부분의 랭킹 사이트는 추천 수 + 서버 수 방식의 리스트를 사용합니다.
가장 큰 예시는 [Minelist](https://minelist.kr) 입니다. 현재 `마인리스트`는 `한국 마인크래프트 서버`들의 동시 접속자 수, 추천 수를 기반으로 순위를 결정합니다.
이 방식은 `이용자 수가 훨씬 많더라도 추천 수가 적으면 순위가 내려가는 문제`를 해결할 수 있으리라 생각됩니다.

랭킹 리스트를 제공하는 서비스에서는 공정성, 형평성이 가장 중요하다고 생각됩니다.
꼭 이 대안들이 아니더라도, 말씀드린 내용을 고려해서 공정성, 형평성을 조율해주셨으면 합니다.

**Comment by wonderlandpark** [comment]
의견 감사합니다. 해당 내용은 회의 안건으로 회의를 거쳐보겠습니다. 혹시 근거로 말씀해주셨던 Minelist의 랭킹이 추천 수로 정렬되는거 같은데 다른 페이지도 있나요??

**Reply by JellyBrick** [reply]
정정합니다. 동시 접속자수 조작이 가능한 문제로 인해 현재는 동접이 적용되지 않는다고 하네요.

**Comment by wonderlandpark** [comment, marked as answer]
안녕하세요, 늦은 답변 죄송합니다.

결론부터 말씀드리자면 현행을 유지해야할 것 같습니다.

가장 큰 이유는 서버 수가 조작될 수 있다는 점 입니다.

또한, 하트 수는 그만큼 해당 봇이 저희 사이트에 기여해주셨다는 의미이기에, 상단에 띄워드리는 것도 합당하다고 생각합니다.

위에 말씀해주셨던 서버 수가 많은 봇에 대해서는 현재는 정비 중이지만 [신뢰된 봇](https://kbots.link/verification)을 확인해주세요.

말씀해주신 의견을 참고하여, 조만간 신뢰된 봇의 혜택 및 노출을 확대토록 하겠습니다.

감사합니다 :)
